libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -232,6 +234,24 @@ vector<Declaration const*> NameAndTypeResolver::cleanedDeclarations(
 	return uniqueFunctions;
 }
 
+void NameAndTypeResolver::warnVariablesNamedLikeInstructions(void)
+{
+	for (auto const& instruction : c_instructions)
+	{
+		string instructionName{instruction.first}; // needs to copy because making it lower case.
+		boost::algorithm::to_lower(instructionName);
+		auto declarations = nameFromCurrentScope(instructionName);
```

Comment:
Right, that might be happening; there might be a warning about `sha3` on every possible source code with inline assembly.  I should add a test about that.